controllers/cluster_controller.go

```diff
@@ -211,10 +214,33 @@ func (r *ClusterReconciler) Reconcile(ctx context.Context, req ctrl.Request) (_
 	}
 
 	defer func() {
+		err := r.updateStatus(ctx, log, cluster)
+		if err != nil {
+			reterr = kerrors.NewAggregate([]error{reterr, err})
+		}
+
+		args := make([]string, 0, len(cluster.Status.Conditions))
+		for _, condition := range cluster.Status.Conditions {
+			args = append(args, fmt.Sprintf("(%s=%s %s)", condition.Type, condition.Status, condition.Reason))
+		}
+
+		log.Info("Current conditions", "conditions", strings.Join(args, ", "))
+
 		// Always attempt to patch the object and status after each reconciliation.
-		if err := patchHelper.Patch(ctx, cluster); err != nil {
+		patchOpts := []patch.Option{}
+
+		// Patch ObservedGeneration only if the reconciliation completed without error
+		if reterr == nil {
+			patchOpts = append(patchOpts, patch.WithStatusObservedGeneration{})
+		}
+		if err := patchCluster(ctx, patchHelper, cluster, patchOpts...); err != nil {
 			reterr = kerrors.NewAggregate([]error{reterr, err})
 		}
+
+		// Only requeue if we are not already re-queueing and the Cluster ready condition is false
```

Comment:
same thing, a bit more about the why would be great
Also, maybe mention we do this to be able to update the status continuously until the cluster becomes ready, since there might be changes in state of the world that don't trigger reconciliation requests